perceval/backend.py

```diff
@@ -121,7 +121,7 @@ def fetch(self, category, **kwargs):
         :returns: a generator of items
         """
         if self.archive:
-            self.archive.init_metadata(self.origin, self.__name__, self.version, category,
+            self.archive.init_metadata(self.origin, self.__class__.__name__, self.version, category,
```

Comment:
To avoid further problems, why don't you create a new test which will check the initialization of an archive object?